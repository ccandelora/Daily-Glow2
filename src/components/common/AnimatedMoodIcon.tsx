import React, { useEffect } from 'react';
import { Animated, StyleSheet, ViewStyle } from 'react-native';
import theme from '../../constants/theme';

interface AnimatedMoodIconProps {
  size?: number;
  color: string;
  style?: ViewStyle;
  children: React.ReactNode;
  active?: boolean;
}

export const AnimatedMoodIcon: React.FC<AnimatedMoodIconProps> = ({
  size = 60,
  color,
  style,
  children,
  active = false,
}) => {
  const scaleAnim = React.useRef(new Animated.Value(1)).current;
  const opacityAnim = React.useRef(new Animated.Value(0.6)).current;

  useEffect(() => {
    Animated.parallel([
      Animated.spring(scaleAnim, {
        toValue: active ? 1.1 : 1,
        useNativeDriver: true,
        damping: 10,
      }),
      Animated.timing(opacityAnim, {
        toValue: active ? 1 : 0.6,
        duration: 200,
        useNativeDriver: true,
      }),
    ]).start();
  }, [active, scaleAnim, opacityAnim]);

  return (
    <Animated.View
      style={[
        styles.container,
        {
          width: size,
          height: size,
          backgroundColor: color,
          transform: [{ scale: scaleAnim }],
          opacity: opacityAnim,
        },
        style,
      ]}
    >
      {children}
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: theme.BORDER_RADIUS.circle,
    alignItems: 'center',
    justifyContent: 'center',
    overflow: 'hidden',
  },
}); 