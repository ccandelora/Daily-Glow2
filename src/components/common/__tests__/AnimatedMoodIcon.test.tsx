import React from 'react';
import { render } from '@testing-library/react-native';
import { Text } from 'react-native';
import { AnimatedMoodIcon } from '../AnimatedMoodIcon';

// Mock Animated
jest.mock('react-native', () => {
  const rn = jest.requireActual('react-native');
  rn.Animated.timing = jest.fn(() => ({
    start: jest.fn(cb => cb && cb({ finished: true })),
  }));
  rn.Animated.spring = jest.fn(() => ({
    start: jest.fn(cb => cb && cb({ finished: true })),
  }));
  rn.Animated.parallel = jest.fn(animations => ({
    start: jest.fn(cb => cb && cb({ finished: true })),
  }));
  return rn;
});

describe('AnimatedMoodIcon', () => {
  it('renders correctly with default props', () => {
    const { toJSON } = render(
      <AnimatedMoodIcon color="#FF5733">
        <Text>ðŸ˜Š</Text>
      </AnimatedMoodIcon>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with custom size', () => {
    const { toJSON } = render(
      <AnimatedMoodIcon color="#FF5733" size={100}>
        <Text>ðŸ˜Š</Text>
      </AnimatedMoodIcon>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when active', () => {
    const { toJSON } = render(
      <AnimatedMoodIcon color="#FF5733" active={true}>
        <Text>ðŸ˜Š</Text>
      </AnimatedMoodIcon>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with custom style', () => {
    const { toJSON } = render(
      <AnimatedMoodIcon 
        color="#FF5733" 
        style={{ marginTop: 10, elevation: 5 }}
      >
        <Text>ðŸ˜Š</Text>
      </AnimatedMoodIcon>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    const { getByText } = render(
      <AnimatedMoodIcon color="#FF5733">
        <Text>Test Child</Text>
      </AnimatedMoodIcon>
    );
    expect(getByText('Test Child')).toBeTruthy();
  });
}); 