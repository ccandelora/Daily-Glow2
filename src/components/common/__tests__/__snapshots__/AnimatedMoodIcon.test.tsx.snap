// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedMoodIcon renders correctly when active 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FF5733",
      "borderRadius": 999,
      "height": 60,
      "justifyContent": "center",
      "opacity": 0.6,
      "overflow": "hidden",
      "transform": [
        {
          "scale": 1,
        },
      ],
      "width": 60,
    }
  }
>
  <Text>
    ðŸ˜Š
  </Text>
</View>
`;

exports[`AnimatedMoodIcon renders correctly with custom size 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FF5733",
      "borderRadius": 999,
      "height": 100,
      "justifyContent": "center",
      "opacity": 0.6,
      "overflow": "hidden",
      "transform": [
        {
          "scale": 1,
        },
      ],
      "width": 100,
    }
  }
>
  <Text>
    ðŸ˜Š
  </Text>
</View>
`;

exports[`AnimatedMoodIcon renders correctly with custom style 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FF5733",
      "borderRadius": 999,
      "elevation": 5,
      "height": 60,
      "justifyContent": "center",
      "marginTop": 10,
      "opacity": 0.6,
      "overflow": "hidden",
      "transform": [
        {
          "scale": 1,
        },
      ],
      "width": 60,
    }
  }
>
  <Text>
    ðŸ˜Š
  </Text>
</View>
`;

exports[`AnimatedMoodIcon renders correctly with default props 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FF5733",
      "borderRadius": 999,
      "height": 60,
      "justifyContent": "center",
      "opacity": 0.6,
      "overflow": "hidden",
      "transform": [
        {
          "scale": 1,
        },
      ],
      "width": 60,
    }
  }
>
  <Text>
    ðŸ˜Š
  </Text>
</View>
`;
